include/motor-control/core/tasks/gear_tmc2160_motor_driver_task.hpp

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include "can/core/can_writer_task.hpp"
+#include "can/core/ids.hpp"
+#include "can/core/messages.hpp"
+#include "common/core/logging.h"
+#include "common/core/message_utils.hpp"
+#include "motor-control/core/stepper_motor/tmc2160.hpp"
+#include "motor-control/core/stepper_motor/tmc2160_driver.hpp"
+#include "motor-control/core/tasks/messages.hpp"
+#include "motor-control/core/tasks/tmc_motor_driver_common.hpp"
+#include "spi/core/messages.hpp"
+
+namespace tmc2160 {
+
+namespace tasks {
+
+namespace gear {
+
+using TaskMessage = tmc::tasks::GearTaskMessage;
+/**
+ * The handler of motor driver messages
+ */
+template <class Writer, can::message_writer_task::TaskClient CanClient,
+          class TaskQueue>
+class MotorDriverMessageHandler {
```

Comment:
Could you use inheritance here to make use of the original 2160 driver task behaviors to avoid repeating code?

Reply:
unfortunately no because then it would accept the other messages. you could make a single class that is templated with the kinds of messages to accept but that's kind of a lot.